plugins/parade-lspcon/fu-parade-lspcon-device.c

```diff
@@ -0,0 +1,829 @@
+/*
+ * Copyright (C) 2021 Peter Marheine <pmarheine@chromium.org>
+ *
+ * SPDX-License-Identifier: LGPL-2.1+
+ */
+
+#include "config.h"
+
+#include <linux/i2c-dev.h>
+
+#include "fu-parade-lspcon-device.h"
+
+/* device registers are split into pages, where
+ * each page has its own I2C address */
+#define I2C_ADDR_PAGE2			0x4A
+#define REG_ADDR_CLT2SPI		0x82
+/* FLASH_ADDR_* are the upper 16 bits of the 24-bit flash address that gets
+ * mapped into page 7. Writing 0x01, 0x42 will map the 256 bytes from 0x420100
+ * into page 7. */
+#define REG_ADDR_FLASH_ADDR_LO		0x8E
+#define REG_ADDR_FLASH_ADDR_HI		0x8F
+/* 16-deep SPI write and read buffer FIFOs */
+#define REG_ADDR_WR_FIFO		0x90
+#define REG_ADDR_RD_FIFO		0x91
+/* Low nibble is write operation length, high nibble for read commands.
+ * Reset to 0 after command completion. */
+#define REG_ADDR_SPI_LEN		0x92
+
+#define REG_ADDR_SPI_CTL		0x93
+/* set to do a write-only transaction */
+#define SPI_CTL_NOREAD			0x04
+/* set to begin executing command */
+#define SPI_CTL_TRIGGER			0x01
+
+/* operation status fields: set to 1 when operation begins, 2 when command has been
+ * sent, reset to 0 when command completed */
+#define REG_ADDR_SPI_STATUS		0x9e
+/* byte programming */
+#define SPI_STATUS_BP_MASK		0x03
+/* sector erase */
+#define SPI_STATUS_SE_MASK		0x0C
+/* chip erase */
+#define SPI_STATUS_CE_MASK		0x30
+
+/* write WR_PROTECT_DISABLE to permit flash write operations */
+#define REG_ADDR_WR_PROTECT		0xB3
+#define WR_PROTECT_DISABLE		0x10
+
+/* MPU control register */
+#define REG_ADDR_MPU			0xBC
+
+/* write a magic sequence to this register to enable writes to
+ * mapped memory via page 7, or anything else to disable */
+#define REG_ADDR_MAP_WRITE		0xDA
+
+#define I2C_ADDR_PAGE5			0x4D
+#define REG_ADDR_ACTIVE_PARTITION	0x0E
+
+#define I2C_ADDR_PAGE7			0x4F
+
+#define FLASH_BLOCK_SIZE		0x10000
+
+/*
+ * user1: 0x10000 - 0x20000
+ * user2: 0x20000 - 0x30000
+ * flag:  0x00002 - 0x00004
+ */
+struct _FuParadeLspconDevice {
+	FuI2cDevice		 parent_instance;
+	guint8			 active_partition;
+	gchar			*aux_device_name;
+};
+
+G_DEFINE_TYPE (FuParadeLspconDevice, fu_parade_lspcon_device, FU_TYPE_I2C_DEVICE)
+
+static void
+fu_parade_lspcon_device_init (FuParadeLspconDevice *self)
+{
+	FuDevice *device = FU_DEVICE (self);
+	fu_device_set_vendor (device, "Parade Technologies");
+	fu_device_add_vendor_id (device, "PCI:0x1AF8");
```

Comment:
Is PCI the right prefix here? I guess it's all we've probably got for Parade...

Reply:
The device name that firmware provides (referred to by the quirks file) is the concatenation of this ID and the device name, so it seems appropriate to include here.